tokio-stream/src/stream_ext.rs

```diff
@@ -993,6 +995,53 @@ pub trait StreamExt: Stream {
     {
         throttle(duration, self)
     }
+
+    /// Collects items into batches inside a vector within a deadline.
+    ///
+    /// Attempts to yield a vector of len `capacity` within the deadline,
+    /// otherwise will yield a vector of len less than `capacity` if the deadline is reached.
+    /// If the wrapped stream does not yield a value before the deadline is reached, then an
+    /// empty vector will be yielded. The deadline will be reset after every batch is yielded.
```

Comment:
If you're looking for inspiration for the documentation, then here is a proposal:

> Batches the items in the given stream using a maximum duration and size for each batch.
> 
> This stream returns the next batch of items in the following situations:
> 
>  1. The inner stream has returned at least `capacity` many items since the last batch.
>  2. The time since the last batch is greater than the given duration.
> 
> The length of the returned vector is never greater than `capacity`. Note that the second condition will trigger even if the resulting batch is empty. In this case, the stream yields an empty vector when the duration has elapsed.

Also, I think `max_size` is a better name than `capacity`.

Reply:
thanks!